### PR TITLE
feat: simple key-> value component for text value type

### LINKED ISSUE #11428: Simple key value component
Create simple key-> value component for mobile where both key and value ans simple  text strings. Key can optionally have a tooltip.

Screenshot 2024-09-25 at 5 23 48 PM

Figma: https://www.figma.com/design/CeAIcptzeaI5ciTpYBaiqr/Confirmations-DS?node-id=4314-18587&node-type=canvas&t=FTDELWNrSVJ25uB6-0

If text is very long it should wrap as below:

Screenshot 2024-09-25 at 5 23 57 PM

### PR DESCRIPTION
**Pull request by jpuri**
## **Description**

Adding simple key-> value component for text value type

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-mobile/issues/11428

## **Manual testing steps**

1. Start storybook locally
2. Check InfoRow component

## **Screenshots/Recordings**
Screenshot 2024-09-30 at 9 00 55 PM

## **Pre-merge author checklist**

- [X] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [X] I've completed the PR template to the best of my ability
- [X] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [X] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/components/Views/confirmations/components/Confirm/Title/Title.tsx [hunk 1]
```diff
@@ -20,11 +20,8 @@ const Title = () => {
   const { approvalRequest } = useApprovalRequest();
   const { colors } = useTheme();
 
+  const title = getTitle(approvalRequest?.type);
   const styles = createStyles(colors);
-  const title = useMemo(
-    () => getTitle(approvalRequest?.type),
-    [approvalRequest?.type],
-  );
```
Comment (jpuri) [inline app/components/Views/confirmations/components/Confirm/Title/Title.tsx:27]: This is too address feedback from previous PR: https://github.com/MetaMask/metamask-mobile/pull/11478#discussion_r1778775603

File: app/components/UI/InfoRow/InfoRow.tsx
Comment (matthewwalsh0) [inline app/components/UI/InfoRow/InfoRow.tsx:1]: Do we want these rows and components under a `confirm` or `confirmations` directory to enforce code ownership?

File: app/components/UI/InfoRow/InfoRow.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import { Text, View } from 'react-native';
+
+import { useTheme } from '../../../util/theme';
+import createStyles from './style';
+
+interface InfoRowProps {
+  label: string;
+  children: React.ReactNode | string;
+  tooltip?: string;
```
Comment (matthewwalsh0) [inline app/components/UI/InfoRow/InfoRow.tsx:10]: Is is intentionally not used yet?

**Review by matthewwalsh0 — APPROVED** [review]

File: app/components/UI/InfoRow/InfoRow.tsx (hunk 2 above)
Comment (jpuri) [reply app/components/UI/InfoRow/InfoRow.tsx:10]: Yep I added this [task](https://github.com/orgs/MetaMask/projects/78/views/1?filterQuery=-label%3Aepic+-status%3A%22%E2%9C%85+Done%22++is%3Aissue+-status%3A%22%E2%9D%8C+Won%27t+do%22&pane=issue&itemId=81603183), unexpected but I found that mobile component lib does not have tooltip.

File: app/components/UI/InfoRow/InfoRow.tsx
Comment (jpuri) [reply app/components/UI/InfoRow/InfoRow.tsx:1]: Not really we can leave them for generic re-use.

Comment (matthewwalsh0) [reply app/components/UI/InfoRow/InfoRow.tsx:1]: My worry is any small changes to these components could have huge impact on the confirmation design without us being aware?

**Review by pedronfigueiredo — APPROVED** [review]

**Review by OGPoyraz — APPROVED** [review]

File: app/components/UI/InfoRow/InfoRow.tsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import { Text, View } from 'react-native';
+
+import { useTheme } from '../../../util/theme';
+import createStyles from './style';
+
+interface InfoRowProps {
+  label: string;
+  children: React.ReactNode | string;
```
[the remark this answers is not in the document]
Comment (jpuri) [reply app/components/UI/InfoRow/InfoRow.tsx:9]: We wrap string in `Text` in this component.

File: app/components/UI/InfoRow/InfoRow.tsx
Comment (jpuri) [reply app/components/UI/InfoRow/InfoRow.tsx:1]: Make sense, I will do it in following PR.

**Pull request merged**